sdk/monitor/Azure.Monitor.OpenTelemetry.Exporter/src/Internals/StandardMetricsExtractionProcessor.cs

```diff
@@ -0,0 +1,102 @@
+using OpenTelemetry.Metrics;
+
+namespace Azure.Monitor.OpenTelemetry.Exporter.Internals
+{
+    internal class StandardMetricsExtractionProcessor : BaseProcessor<Activity>
+    {
+        private bool _disposed;
+        private AzureMonitorResource _resource;
+        private readonly Meter _meter;
+        private readonly Histogram<double> _requestDuration;
+        private readonly MeterProvider _meterprovider;
+
+        internal static readonly IReadOnlyDictionary<string, string> s_standardMetricNameMapping = new Dictionary<string, string>()
+        {
+            [StandardMetricConstants.RequestDurationInstrumentName] = StandardMetricConstants.RequestDurationMetricIdValue,
+        };
+
+        internal MeterProvider StandardMetricMeterProvider { get { return _meterprovider; } }
+
+        internal AzureMonitorResource StandardMetricResource => _resource ??= ParentProvider.GetResource().UpdateRoleNameAndInstance();
+
+        internal StandardMetricsExtractionProcessor(MetricReader standardMetricReader)
+        {
+            _meter = new Meter(StandardMetricConstants.StandardMetricMeterName);
+            _requestDuration = _meter.CreateHistogram<double>(StandardMetricConstants.RequestDurationInstrumentName);
+
+            _meterprovider = Sdk.CreateMeterProviderBuilder()
+                .AddMeter(StandardMetricConstants.StandardMetricMeterName)
+                .AddReader(standardMetricReader)
+                .Build();
+        }
+
+        public override void OnEnd(Activity activity)
+        {
+            if (activity.Kind == ActivityKind.Server)
+            {
+                activity.SetTag("_MS.ProcessedByMetricExtractors", "(Name: X,Ver:'1.1')");
+                ReportRequestDurationMetric(activity, SemanticConventions.AttributeHttpStatusCode);
+            }
+
+            // TODO: other activity kinds
+        }
+
+        private void ReportRequestDurationMetric(Activity activity, string statusCodeAttribute)
+        {
+            string statusCodeAttributeValue = null;
+            foreach (var tag in activity.EnumerateTagObjects())
+            {
+                if (tag.Key == statusCodeAttribute)
+                {
+                    statusCodeAttributeValue = tag.Value.ToString();
+                    break;
+                }
+            }
+
+            TagList tags = default;
+            tags.Add(new KeyValuePair<string, object>(StandardMetricConstants.RequestResultCodeKey, statusCodeAttributeValue));
+            tags.Add(new KeyValuePair<string, object>(StandardMetricConstants.MetricIdKey, StandardMetricConstants.RequestDurationMetricIdValue));
+            tags.Add(new KeyValuePair<string, object>(StandardMetricConstants.IsAutoCollectedKey, "True"));
+            tags.Add(new KeyValuePair<string, object>(StandardMetricConstants.IsSyntheticKey, "False"));
+            tags.Add(new KeyValuePair<string, object>(StandardMetricConstants.CloudRoleInstanceKey, StandardMetricResource.RoleInstance));
```

Comment:
Most of these are common and does not change in every execution. Should we consider creating once?

Reply:
We could. However, only two of these will remain constant.

```csharp
tags.Add(new KeyValuePair<string, object>(StandardMetricConstants.MetricIdKey, StandardMetricConstants.RequestDurationMetricIdValue));
tags.Add(new KeyValuePair<string, object>(StandardMetricConstants.IsAutoCollectedKey, "True"));
```

Reply:
for the constant one, model as resource.

Reply:
rolename and roleNameInstance may not change.

Reply:
As we discussed, this could be refactored in a different PR.